src/prism.c

```diff
@@ -10328,6 +10328,11 @@ parse_write_name(pm_parser_t *parser, pm_constant_id_t *name_field) {
 
     // Now switch the name to the new string.
     *name_field = pm_constant_pool_insert_owned(&parser->constant_pool, name, length + 1);
+
+    #ifdef __clang_analyzer__
```

Comment:
I'm not sure how I feel about this addition, because it would actually case a double free. It's definitely going to be freed by the constant pool, so we're introducing a bug here just to satisfy the analyzer. Is there a way to put this into a config file instead?

Reply:
I couldn't find another way. I see usages of the symbol `__clang_analyzer__` in other projects so I think maybe it is OK. But I definitely agree with you on this may not be an ideal solution.

Reply:
OK, it seems like adding `// NOLINT` to the reported line also silences these warnings.